src/pages/trade/store/positions.ts

```diff
@@ -0,0 +1,99 @@
+import { makeAutoObservable } from 'mobx';
+import { TransactionPlannerRequest } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+import { PositionId } from '@penumbra-zone/protobuf/penumbra/core/component/dex/v1/dex_pb';
+import { planBuildBroadcast } from '@/pages/trade/ui/order-form/helpers.tsx';
+import { connectionStore } from '@/shared/model/connection';
+import { AddressIndex } from '@penumbra-zone/protobuf/penumbra/core/keys/v1/keys_pb';
+import { queryClient } from '@/shared/const/queryClient.ts';
+import { getPositionData, PositionData } from '@/pages/trade/api/positions.ts';
+import { penumbra } from '@/shared/const/penumbra.ts';
+import { DexService } from '@penumbra-zone/protobuf';
+import { openToast } from '@penumbra-zone/ui/Toast';
+
+class PositionsStore {
+  public loading = false;
+
+  constructor() {
+    makeAutoObservable(this);
+  }
+
+  setLoading(state: boolean) {
+    this.loading = state;
+  }
```

Comment:
comment: I've been reading about https://medium.com/@brojd/automate-loading-statuses-in-mobx-d6965ad738d9 and https://mobx.js.org/actions.html#using-flow-instead-of-async--await- for automated loading and using `flow` annotation instead of async / await for mobx state management. this might be something interesting to look into the future?

Reply:
Interesting! No idea, but let's continue to explore good async patterns like this 👍